PubSub/Sources/API/SubscriptionsAPI.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  SubscriptionsAPI.swift
+//  
+//
+//  Created by Susheel Athmakuri on 6/21/21.
+//
+
+import Core
+import NIO
+import AsyncHTTPClient
+import Foundation
+
+public protocol SubscriptionsAPI {
+    func get(subscriptionId: String) -> EventLoopFuture<GoogleCloudPubSubSubscription>
```

Comment:
Can we add docs/comments about the parameters for using the API? The models aren't necessarily too important at the moment but the actual API calls would be useful.

Reply:
Done.